### PR TITLE
[SmartSwitch] Add implementation for the DPU chassis daemon.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**

### CODE REVIEW

File: sonic-chassisd/scripts/chassisd [hunk 1]
```diff
@@ -687,14 +791,42 @@ class ChassisdDaemon(daemon_base.DaemonBase):
 
         self.log_info("Shutting down...")
 
+
+class DpuChassisdDaemon(ChassisdDaemon):
```
Comment (vivekrnv) [inline sonic-chassisd/scripts/chassisd:794]: Only reason to inherit from ChassisDaemon to leverage signal handler correct?

File: sonic-chassisd/scripts/chassisd [hunk 2]
```diff
@@ -605,58 +612,155 @@ class ConfigManagerTask(ProcessTaskBase):
...
+
+#
+# State Manager task ========================================================
+#
+
+class DpuStateUpdater(logger.Logger):
+
+    DP_STATE = 'dpu_data_plane_state'
+    DP_UPDATE_TIME = 'dpu_data_plane_time'
+    CP_STATE = 'dpu_control_plane_state'
```
Comment (vivekrnv) [inline sonic-chassisd/scripts/chassisd:624]: CP_STATE is defined twice. #Resolved

File: sonic-chassisd/scripts/chassisd [hunk 3]
```diff
@@ -605,58 +612,155 @@ class ConfigManagerTask(ProcessTaskBase):
...
+            self._get_dp_state = self.chassis.get_dataplane_state
+
+        try:
+            self.chassis.get_controlplane_state()
+        except NotImplementedError:
+            self._get_cp_state = self._get_control_plane_state_common
+        else:
+            self._get_cp_state = self.chassis.get_controlplane_state
+
+        self.id = self.chassis.get_dpu_id()
```
Comment (vivekrnv) [inline sonic-chassisd/scripts/chassisd:653]: Will the self.id be 0 for NPU?

File: sonic-chassisd/scripts/chassisd [hunk 4]
```diff
@@ -605,58 +612,155 @@ class ConfigManagerTask(ProcessTaskBase):
...
+        self.name = f'DPU{self.id}'
+
+        self.dpu_state_table = swsscommon.Table(self.chassis_state_db, 'DPU_STATE')
+
+    def _get_data_plane_state_common(self):
+        port_table = swsscommon.Table(self.app_db, 'PORT_TABLE')
+
+        for port in self.config_db.get_table('PORT'):
```
Comment (vivekrnv) [inline sonic-chassisd/scripts/chassisd:661]: Why read from config_db anyway. Might as well loop over PORT_TABLE in APPL_DB to verify the oper status

File: sonic-chassisd/scripts/chassisd (hunk 1 above)
Comment (oleksandrivantsiv) [reply sonic-chassisd/scripts/chassisd:794]: the reason is to inherit the interface of `DaemonBase`

File: sonic-chassisd/scripts/chassisd (hunk 3 above)
Comment (oleksandrivantsiv) [reply sonic-chassisd/scripts/chassisd:653]: `DpuStateUpdater` will be created only when chassis is running on the DPU. For the NPU this code won't be executed.

File: sonic-chassisd/scripts/chassisd (hunk 4 above)
Comment (oleksandrivantsiv) [reply sonic-chassisd/scripts/chassisd:661]: `PORT_TABLE` contains extra entries lile `port_init_done` and others. So get a list of ports we need to filter by port name prefix. I don't want to hardcode the name prefix and instead getting a list of ports dynamically from the config DB

File: sonic-chassisd/scripts/chassisd (hunk 2 above)
Comment (oleksandrivantsiv) [reply sonic-chassisd/scripts/chassisd:624]: removed

File: sonic-chassisd/scripts/chassisd (hunk 1 above)
Comment (vivekrnv) [reply sonic-chassisd/scripts/chassisd:794]: Why can't we inherit from daemon_base.DaemonBase directly?

File: sonic-chassisd/scripts/chassisd (hunk 4 above)
Comment (vivekrnv) [reply sonic-chassisd/scripts/chassisd:661]: I see, okay. #Resolved

File: sonic-chassisd/scripts/chassisd (hunk 3 above)
Comment (vivekrnv) [reply sonic-chassisd/scripts/chassisd:653]: Looks like it's also created for DPU

```
    if chassis.is_smartswitch() and chassis.is_dpu():
        chassisd = DpuChassisdDaemon(SYSLOG_IDENTIFIER, chassis)
    else:
        chassisd = ChassisdDaemon(SYSLOG_IDENTIFIER, chassis)
```

Comment (vivekrnv) [reply sonic-chassisd/scripts/chassisd:653]: Got it, #Resolved

**Review by vivekrnv — APPROVED** [review]

**Pull request closed without merging**